Moving documentation to ReadTheDocs

mkavulich opened the pull request:
This PR removes the old, LaTeX-based users guide in favor of RST files that are buildable with Sphinx. These new docs have been configured to automatically build with each commit to `main` on ReadTheDocs, and can be viewed here: [html](https://ccpp-scm.readthedocs.io/en/latest/) | [PDF](https://ccpp-scm.readthedocs.io/_/downloads/en/latest/pdf/)

For limitations in the new docs compared to the previous LaTeX/PDF, see [this comment](https://github.com/NCAR/ccpp-scm/issues/418#issuecomment-1991816569)

**NOTE:** The original LaTeX files were auto-converted from `.tex` to `.rst` using the `pandoc` utility, and then manually updated for format. **The conversion was not perfect:** in particular, most section/chapter links were broken, and all `code-case text` was omitted from the converted docs. These problems needed to be fixed manually, but it's entirely possible some of these may have slipped into the final product.

Resolves #418

scrasmussen commented:
@mkavulich this looks really great! I checked out the branch to add documentation for the single precision build and it was easy to use and worked well. A few comments:
- with the move away from LaTeX it seems the [Makefile](https://github.com/mkavulich/ccpp-scm/blob/feature/convert_doc_to_readthedocs/scm/doc/TechGuide/Makefile) should be updated or deleted?
- I don't know if would be helpful if there was a small section in the documentation to tell users how to build the readthedocs locally or if it should be a `README.md` under `scm/doc/TechGuide`? Apologies if this information is already in the docs somewhere. I installed the [requirements](https://github.com/mkavulich/ccpp-scm/blob/feature/convert_doc_to_readthedocs/scm/doc/TechGuide/requirements.txt) and then ended up googling the command `$ sphinx-build -M html [source_dir] [output_dir]`
- should the [chap_function.rst](https://github.com/mkavulich/ccpp-scm/blob/feature/convert_doc_to_readthedocs/scm/doc/TechGuide/chap_function.rst) file be named `chap_algorithm.rst` since it is the Algorithm chapter? 

Anyway, these are just small things, overall it worked very well for me!

mkavulich commented:
@scrasmussen Thanks for the review! I have updated the Makefile, though I am unable to test the latex/pdf rule because I can not install latexmk on my machine (which is part of why this whole conversation of moving to ReadTheDocs started!). I also added a README.md file to assist people who are interested in building the HTML or PDF documentation.

With regards to renaming documents/chapters, I kept the names from the original `.tex` files. I would rather keep the format-transition commits contained here and make any other cleanup in subsequent PRs.

scrasmussen reviewed (approved):
Everything builds and looks good!

hertneky commented on scm/doc/TechGuide/chap_function.rst:
```diff
@@ -0,0 +1,157 @@
+
+The following steps are performed at the beginning of program execution:
+
+#. Call ``get_config_nml()`` in the ``scm_input`` module to read in the ``case_config`` and
```
case_config was linked in the original version to the case_config sec

mkavulich replied:
Unfortunately one of the limits of Sphinx is that it is unable to combine links with code formatting. I have updated this to add a link as a compromise.

hertneky commented on scm/doc/TechGuide/chap_function.rst:
```diff
@@ -0,0 +1,157 @@
+   read.
+
+#. Call ``get_case_init()`` (or ``get_case_init_DEPHY()`` if using the DEPHY format) in the ``scm_input`` module to read in the
+   :numref:`case input data file <case input>`. This subroutine
```
this isn't getting linked to the case input data file sec properly

hertneky commented on scm/doc/TechGuide/chap_function.rst:
```diff
@@ -0,0 +1,157 @@
+output periods:
+
+#. one associated with how often instantaneous variables should be
+   written out (controlled by the ``-- n_itt_out`` run script variable).
```
extra space --n_itt_out

hertneky commented on scm/doc/TechGuide/chap_function.rst:
```diff
@@ -0,0 +1,157 @@
+   written out (controlled by the ``-- n_itt_out`` run script variable).
+
+#. one associated with how often diagnostic (either instantaneous or
+   time-averaged) should be written out (controlled by the ``-- n_itt_diag`` run script
```
extra space again

hertneky commented on scm/doc/TechGuide/chap_ccpp.rst:
```diff
@@ -0,0 +1,383 @@
+         .. code:: fortran
+
+            call check(NF90_INQ_VARID(grp_ncid,"smoke",varID))
+                    call check(NF90_GET_VAR(grp_ncid,varID,input_smoke))
```
extra indent

hertneky replied:
I still see indent here

hertneky commented on scm/doc/TechGuide/chap_ccpp.rst:
```diff
@@ -0,0 +1,383 @@
+            do i=1, scm_state%n_cols
+                            call interpolate_to_grid_centers(scm_input%input_nlev, scm_input%input_pres, scm_input%input_smoke, scm_state%pres_l(i,1,:), &
+                                    scm_state%n_levels, scm_state%state_tracer(i,1,:,scm_state%smoke_index,1), last_index_init, 1)
+                    end do
```
Indentation on this block is off

hertneky replied:
Indentation here still off

mkavulich replied:
Sorry about that, somehow I missed these requested indentation changes. Fixed now!

hertneky commented on scm/doc/TechGuide/chap_hsd.rst:
```diff
@@ -0,0 +1,352 @@
+================================
+
+Chapter 7 of the CCPP v6 Technical Documentation
+(https://ccpp-techdoc.readthedocs.io/en/v6.0.0/) provides an overview of
```
Chapter 7 of this CCPP v6 tech doc seems to be Code Management?

mkavulich replied:
This section was added by @dustinswales back when he added the Suite Simulator documentation. I think this is actually supposed to be referencing this chapter itself, not the Tech Doc. I updated it accordingly.

hertneky replied:
I see this as "Chapter %c provides an overview...."

mkavulich replied:
Should be fixed now

hertneky reviewed (changes requested):
Looks good - just a handful of things I noticed in terms of formatting.

mkavulich reviewed (commented):
@hertneky Thank you for your thorough review! I believe I have addressed all of your comments.

hertneky reviewed (approved):
Looks good now. Thanks Mike

grantfirl commented:
I didn't review in detail, although, in general, it looks great. Although I love LaTeX, this is a better way to format software docs in pretty much every way. Thanks for doing this.
